### ISSUE TITLE
(choco-package-list-backup) Allow choice of OneDrive Locations

### ISSUE THREAD

**Issue by snowdej** [bcurran3/ChocolateyPackages_CCR#78]
Allow the XML config to include all three onedrive env variables

Existing
OneDrive

new Ones  in xml
```
    <UseOneDriveCommercial>false</UseOneDriveCommercial>
    <UseOneDriveConsumer>false</UseOneDriveConsumer>
```

ps1 file
Add the following 
```
$UseOneDriveCommercial = $ConfigFile.Settings.Preferences.UseOneDriveCommercial
$UseOneDriveConsumer = $ConfigFile.Settings.Preferences.UseOneDriveConsumer

   # Backup Chocolatey package names on local computer to packages.config file in OneDriveCommercial directory if it exists
if ($ENV:OneDriveCommercial) {$OneDriveCommercialExists="True"} else {$OneDriveCommercialExists="False"}
if ($UseOneDriveCommercial -match "True" -and ($OneDriveCommercialExists -match "True"))
   {
    $SavePath = "$ENV:OneDriveCommercial\$SaveFolderName\$ENV:ComputerName"
    Write-PackagesConfig
   }      
   
   # Backup Chocolatey package names on local computer to packages.config file in OneDriveConsumer directory if it exists
if ($ENV:OneDriveConsumer) {$OneDriveConsumerExists="True"} else {$OneDriveConsumerExists="False"}
if ($UseOneDriveConsumer -match "True" -and ($OneDriveConsumerExists -match "True"))
   {
    $SavePath = "$ENV:OneDriveConsumer\$SaveFolderName\$ENV:ComputerName"
    Write-PackagesConfig
   }      

```

**Comment by bcurran3** [comment]
Thanks.

Funny thing is old versions of the package used the consumer environmental variable but I changed it for someone who moved the directory from default. I'll be glad to add in the paid version support soon.

**Comment by snowdej** [comment]
I think what happens is onedriveconsumer and onesriveCommercial are set by the installer of each... But both update the OneDrive env variable - so which ever version you installed last will set the onedrive variable in addition to the consumer/commercial one of its own

**Comment by bcurran3** [comment]
Thanks. Some research needs to be done to verify. Some regkey checking would probably determine as well.

------------------------------------
I just checked the code. Seems it was not OneDrive that switched from the environmental variable:

```
# Backup Chocolatey package names on local computer to packages.config file in OneDrive directory if it exists
if ($ENV:OneDrive) {$OneDriveExists="True"} else {$OneDriveExists="False"}
if ($UseOneDrive -match "True" -and ($OneDriveExists -match "True"))
   {
    $SavePath = "$ENV:OneDrive\$SaveFolderName\$ENV:ComputerName"
    Write-PackagesConfig
   }    
```

**Comment by bcurran3** [comment]
A quick look at my $ENV:LocalAppData\OneDrive folder shows a settings folder with a Personal in it. Can you verify if there is a Commercial/Pro or some such folder in yours? This could make it easy to check for both.

**Comment by snowdej** [comment]
Hi

Oddly no

in appdata there is just one folder for onedrive - nothing in it mind

C:\Users\julian\AppData\Local\OneDrive

The actual onedrive folders are in  C:\Users\julian\

This is the list of env variable powershell sees called one*

PS C:\Users\julian>  Get-ChildItem Env:One*

Name                           Value
----                           -----
OneDriveConsumer               C:\Users\julian\OneDrive
OneDrive                       C:\Users\julian\OneDrive
OneDriveCommercial             C:\Users\julian\OneDrive - xxx Ltd

regards

Julian

**Comment by bcurran3** [comment]
Julian,

Thanks for the quick responses.

Not what I expected, but good to know, points me in the right direction!

I'm going to most likely bug you for the regkeys later...

**Comment by jeff9315** [comment]
Since I don't use OneDrive, I don't know if this is feasible ... but maybe a good solution is to allow multiple custom locations. Then one could be for Enterprise One Drive, one for personal One Drive, one for a custom Google Drive location, etc.

**Comment by bcurran3** [comment]
@jeff9315 - Multiple custom locations is easy and reasonable. Create a new issue if you'd like this done please.

As for the multiple OneDrive locations - I'll most likely just end up checking for all three of the environmental variables and use them all if found.

**Comment by bcurran3** [comment]
I just added those other locations based on environmental variable availability. 

I did not change the config options to granularly separate out based on subscription types and am leaving them all under one OneDrive grouping. My thought is that if you have personal and corporate subscriptions, it's likely desirable to backup the package list to both places.

Untested but should work fine. Added to v2019.07.02
